src/main/java/org/cbioportal/application/file/utils/ZipOutputStreamWriterFactory.java

```diff
@@ -0,0 +1,77 @@
+package org.cbioportal.application.file.utils;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public class ZipOutputStreamWriterFactory implements FileWriterFactory, Closeable {
+
+  private final OutputStream outputStream;
+  private final ZipOutputStream zipOutputStream;
```

Comment:
Curious if you tested out different Compression Levels?

Reply:
no, I didn't. Do you think we should use non-default compression level instead?